### DISCUSSION TITLE
Nuxt3: how to handle 404 errors with [..slug].vue and static generation?

### DISCUSSION THREAD

**Discussion by codeflorist** [nuxt/nuxt#16704] [General]
i'm at a loss here on how to get a 404 error page working with a statically generated Nuxt 3 app, that uses catch-all route via `[...slug].vue`.

i believe throwing an error and using a `error.vue` would be the way to go.

here would be my `[...slug].vue`:
```vue
<script setup>
    const route = useRoute()
    const apiRoutes = ['/','/foo','/bar']
    if (!apiRoutes.includes(route.path)) {
        throw createError({ statusCode: 404, statusMessage: 'Page Not Found' })
    }
</script>

<template>
    <div>
        <NuxtLink to="/foo">foo</NuxtLink>
        <br />
        <NuxtLink to="/bar">bar</NuxtLink>
    </div>
</template>
```

within a dev-environment, calling an invalid route correctly shows the content of my `error.vue`.

this however doesn't seem to work on generated static pages, since then the error is thrown in the console and a white page is displayed. is this a nuxt-bug? should error.vue work with `nuxt generate`?

if not, what would be the way to go for handling 404s in this case?

**Comment by danielroe** [comment]
Add `fatal: true` to that error.

**Reply by codeflorist** [reply]
that did the trick. thanks a lot!
